// @sil/color/dist/types.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ColorType = exports.instanceOfCMYK = exports.instanceOfHSVA = exports.instanceOfHSV = exports.instanceOfHSLA = exports.instanceOfHSL = exports.instanceOfRGBA = exports.instanceOfRGB = void 0;
function instanceOfRGB(obj) {
    return ("r" in obj &&
        "g" in obj &&
        "b" in obj &&
        !("a" in obj) &&
        Object.keys(obj).length == 3);
}
exports.instanceOfRGB = instanceOfRGB;
function instanceOfRGBA(obj) {
    return ("r" in obj &&
        "g" in obj &&
        "b" in obj &&
        "a" in obj &&
        Object.keys(obj).length == 4);
}
exports.instanceOfRGBA = instanceOfRGBA;
function instanceOfHSL(obj) {
    return ("h" in obj &&
        "s" in obj &&
        "l" in obj &&
        !("a" in obj) &&
        Object.keys(obj).length == 3);
}
exports.instanceOfHSL = instanceOfHSL;
function instanceOfHSLA(obj) {
    return ("h" in obj &&
        "s" in obj &&
        "l" in obj &&
        "a" in obj &&
        Object.keys(obj).length == 4);
}
exports.instanceOfHSLA = instanceOfHSLA;
function instanceOfHSV(obj) {
    return ("h" in obj &&
        "s" in obj &&
        "v" in obj &&
        !("a" in obj) &&
        Object.keys(obj).length == 3);
}
exports.instanceOfHSV = instanceOfHSV;
function instanceOfHSVA(obj) {
    return ("h" in obj &&
        "s" in obj &&
        "v" in obj &&
        "a" in obj &&
        Object.keys(obj).length == 4);
}
exports.instanceOfHSVA = instanceOfHSVA;
function instanceOfCMYK(obj) {
    return ("c" in obj &&
        "m" in obj &&
        "y" in obj &&
        "k" in obj &&
        Object.keys(obj).length == 4);
}
exports.instanceOfCMYK = instanceOfCMYK;
exports.ColorType = {
    RGB: "rgb",
    RGBA: "rgba",
    HSL: "hsl",
    HSLA: "hsla",
    HEX: "hex",
    HSV: "hsv",
    HSVA: "hsva",
    CMYK: "cmyk",
    UNKNOWN: "unknown",
};
