// cli-block/dist/test/kitchensink.js
"use strict";
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
const blocks_1 = require("../lib/blocks");
const border_model_1 = require("../lib/border/border.model");
const settings_1 = require("../lib/settings/");
let cliSettings = {
    borderColor: border_model_1.BorderColor.yellow,
    borderType: border_model_1.BorderType.fat,
};
const JSON_EXAMPLE = {
    String: "is a test",
    Number: 2000,
    Object: { item1: "value1", item2: "value1" },
    Array: ["item1", "item2"],
    ObjectArray: [{ item1: "value1" }, { item2: "value2" }],
};
const TABLE_EXAMPLE = [
    ["Type", "Default", "Description"],
    ["Boolean", false, "Does work", "item5"],
    ["String", "Some item", "Yes, or no"],
    ["Number", 5, "How much?"],
];
const show = () => __awaiter(void 0, void 0, void 0, function* () {
    (0, blocks_1.start)("The Script Title", cliSettings);
    (0, blocks_1.blockHeader)("hoi");
    (0, blocks_1.blockLine)();
    (0, blocks_1.blockLine)("with content");
    (0, blocks_1.blockMid)();
    (0, blocks_1.blockMid)("JSON");
    yield (0, blocks_1.blockJson)(JSON_EXAMPLE);
    (0, blocks_1.blockMid)("Settings");
    yield (0, blocks_1.blockSettings)(JSON_EXAMPLE);
    (0, blocks_1.blockMid)("default settings");
    yield (0, blocks_1.blockSettings)(settings_1.defaultSettings);
    (0, blocks_1.blockMid)("default table");
    yield (0, blocks_1.blockTable)(TABLE_EXAMPLE);
    (0, blocks_1.blockMid)("Full width, headerless table");
    yield (0, blocks_1.blockTable)(TABLE_EXAMPLE, {
        tableHeader: false,
        tableSpace: false,
        padding: 0,
    });
    (0, blocks_1.blockMid)("Specials");
    (0, blocks_1.blockLineError)("An error");
    (0, blocks_1.blockLineWarning)("A warning");
    (0, blocks_1.blockLineSuccess)("Yes, this works");
    (0, blocks_1.blockMid)("Content");
    (0, blocks_1.blockLine)(`Fusce interdum blandit ligula, eu ornare tellus convallis ut. Praesent ut elit tempor, luctus urna ac, mattis est. Donec tincidunt sollicitudin eleifend. Donec eu rutrum arcu. Pellentesque lobortis ante libero, vel ultrices mi cursus at. Nullam et leo in justo lacinia blandit in eu quam. Donec magna lectus, posuere sed urna sit amet, venenatis vehicula libero. Donec sed sagittis elit. Suspendisse at dolor vel ipsum aliquet pulvinar sit amet accumsan sapien. Vivamus eget odio mi. Cras tempus vel sapien non viverra. Proin id placerat ipsum. In cursus nisi vel nulla bibendum finibus. Curabitur consequat tortor at tellus rutrum, vel rhoncus magna tempus. Quisque et nisl ac magna cursus aliquet. Donec eget luctus diam. Nulla et lectus lacinia, faucibus risus eget, eleifend neque. Aenean ut luctus est. Mauris dui libero, dapibus vel vehicula quis, convallis vitae neque. Nunc faucibus lacus urna, a accumsan nulla dapibus in. Duis suscipit augue sed magna efficitur, id imperdiet arcu gravida. Cras luctus, eros sed varius luctus, erat urna ullamcorper risus, non scelerisque felis velit luctus sem. Quisque arcu ipsum, vestibulum quis efficitur pretium, hendrerit ac eros. Curabitur at ligula sed mauris pretium euismod. Donec accumsan nisi quis sapien elementum consectetur.`);
    (0, blocks_1.blockMid)("Count down");
    yield (0, blocks_1.blockCounter)({
        message: "Still [count] seconds to go...",
        start: 5,
        end: 0,
        increment: 1,
        interval: 250,
    });
    (0, blocks_1.blockMid)("Count up");
    yield (0, blocks_1.blockCounter)({
        message: "Loading the end [count]%",
        start: 0,
        end: 100,
        increment: 1,
        interval: 25,
    });
    (0, blocks_1.blockMid)("Count up, multi message");
    yield (0, blocks_1.blockCounter)({
        messages: [
            `Starting with [count]`,
            `And this is [count]`,
            `still [count] to go`,
            `now about [count]`,
            `what a joy..:)`,
            `finally [count]`,
        ],
        start: 0,
        end: 5,
        increment: 1,
        interval: 250,
    });
    (0, blocks_1.blockMid)("Count down, multi message");
    yield (0, blocks_1.blockCounter)({
        messages: [
            `Starting with [count]`,
            `And this is [count]`,
            `still [count] to go`,
            `now about [count]`,
            `what a joy..:)`,
            `finally [count]`,
        ],
        start: 5,
        end: 0,
        increment: 1,
        interval: 250,
    });
    (0, blocks_1.blockMid)("Loader");
    yield (0, blocks_1.blockLoader)();
    (0, blocks_1.blockMid)("Loader Custom");
    yield (0, blocks_1.blockLoader)({
        message: "[loader] [percentage] to go",
        increment: 2,
        width: "50",
        start: 0,
        end: 100,
        interval: 50,
        charFilled: "╱",
        charUnfilled: "╲",
    });
    let i = 0;
    const loaderInterval = setInterval(() => __awaiter(void 0, void 0, void 0, function* () {
        yield (0, blocks_1.blockStepLoader)({
            message: "[loader] [percentage] to go",
            width: "50",
            start: 0,
            end: 100,
            step: i,
        });
        i++;
        if (i - 1 == 100) {
            clearInterval(loaderInterval);
        }
    }), 25);
    (0, blocks_1.blockFooter)();
});
const full = () => __awaiter(void 0, void 0, void 0, function* () {
    (0, blocks_1.blockFull)(`Fusce interdum blandit ligula, eu ornare tellus convallis ut. Praesent ut elit tempor, luctus urna ac, mattis est. Donec tincidunt sollicitudin eleifend. Donec eu rutrum arcu. Pellentesque lobortis ante libero, vel ultrices mi cursus at. Nullam et leo in justo lacinia blandit in eu quam. Donec magna lectus, posuere sed urna sit amet, venenatis vehicula libero. Donec sed sagittis elit. Suspendisse at dolor vel ipsum aliquet pulvinar sit amet accumsan sapien. Vivamus eget odio mi. Cras tempus vel sapien non viverra. Proin id placerat ipsum. In cursus nisi vel nulla bibendum finibus. Curabitur consequat tortor at tellus rutrum, vel rhoncus magna tempus. Quisque et nisl ac magna cursus aliquet. Donec eget luctus diam. Nulla et lectus lacinia, faucibus risus eget, eleifend neque. Aenean ut luctus est. Mauris dui libero, dapibus vel vehicula quis, convallis vitae neque. Nunc faucibus lacus urna, a accumsan nulla dapibus in. Duis suscipit augue sed magna efficitur, id imperdiet arcu gravida. Cras luctus, eros sed varius luctus, erat urna ullamcorper risus, non scelerisque felis velit luctus sem. Quisque arcu ipsum, vestibulum quis efficitur pretium, hendrerit ac eros. Curabitur at ligula sed mauris pretium euismod. Donec accumsan nisi quis sapien elementum consectetur.`, {}, { header: "Test" });
    (0, blocks_1.blockFull)("Fusce interdum blandit ligula, eu ornare tellus convallis ut. Praesent ut elit tempor, luctus urna ac, mattis est. Donec tincidunt sollicitudin eleifend. Donec eu rutrum arcu. Pellentesque lobortis ante libero, vel ultrices mi cursus at");
});
(() => __awaiter(void 0, void 0, void 0, function* () {
    yield show();
    full();
}))();
