// cli-block/dist/lib/legacy.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.CREATE_BLOCK_TABLE = exports.BLOCK_TABLE = exports.CREATE_BLOCK_SETTINGS = exports.BLOCK_SETTINGS = exports.CREATE_BLOCK_ROW_LINE = exports.BLOCK_ROW_LINE = exports.CREATE_BLOCK_LINE = exports.BLOCK_LINE = exports.CREATE_BLOCK_JSON = exports.BLOCK_JSON = exports.BLOCK_ERRORS = exports.BLOCK_WARNINGS = exports.CREATE_BLOCK_LINE_ERROR = exports.BLOCK_LINE_ERROR = exports.CREATE_BLOCK_LINE_WARNING = exports.BLOCK_LINE_WARNING = exports.CREATE_BLOCK_LINE_SUCCESS = exports.BLOCK_LINE_SUCCESS = exports.CREATE_BLOCK_FOOTER = exports.BLOCK_FOOTER = exports.CREATE_BLOCK_START = exports.BLOCK_START = exports.CREATE_BLOCK_END = exports.BLOCK_END = exports.BLOCK_LOADER = exports.BLOCK_COUNTER = exports.CREATE_BLOCK = exports.START = exports.BLOCK = void 0;
const blocks = __importStar(require("./blocks"));
exports.BLOCK = blocks.block;
exports.START = blocks.start;
exports.CREATE_BLOCK = blocks.createBlock;
exports.BLOCK_COUNTER = blocks.blockCounter;
exports.BLOCK_LOADER = blocks.block;
exports.BLOCK_END = blocks.blockFooter;
exports.CREATE_BLOCK_END = blocks.createBlockFooter;
exports.BLOCK_START = blocks.blockHeader;
exports.CREATE_BLOCK_START = blocks.createBlockHeader;
exports.BLOCK_FOOTER = blocks.blockFooter;
exports.CREATE_BLOCK_FOOTER = blocks.createBlockFooter;
exports.BLOCK_LINE_SUCCESS = blocks.blockLineSuccess;
exports.CREATE_BLOCK_LINE_SUCCESS = blocks.createBlockLineSuccess;
exports.BLOCK_LINE_WARNING = blocks.blockLineWarning;
exports.CREATE_BLOCK_LINE_WARNING = blocks.createBlockLineWarning;
exports.BLOCK_LINE_ERROR = blocks.blockLineError;
exports.CREATE_BLOCK_LINE_ERROR = blocks.createBlockLineError;
exports.BLOCK_WARNINGS = blocks.blockWarnings;
exports.BLOCK_ERRORS = blocks.blockErrors;
exports.BLOCK_JSON = blocks.blockJson;
exports.CREATE_BLOCK_JSON = blocks.createBlockJson;
exports.BLOCK_LINE = blocks.blockLine;
exports.CREATE_BLOCK_LINE = blocks.createBlockLine;
exports.BLOCK_ROW_LINE = blocks.blockRowLine;
exports.CREATE_BLOCK_ROW_LINE = blocks.createBlockRowLine;
exports.BLOCK_SETTINGS = blocks.blockSettings;
exports.CREATE_BLOCK_SETTINGS = blocks.createBlockSettings;
exports.BLOCK_TABLE = blocks.blockTable;
exports.CREATE_BLOCK_TABLE = blocks.createBlockTable;
