// cli-block/dist/lib/util/style.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.color = exports.gray = exports.white = exports.black = exports.magenta = exports.dim = exports.bold = exports.yellow = exports.italic = exports.green = exports.blue = exports.red = void 0;
const color_1 = require("@sil/color");
var kleur_1 = require("kleur");
Object.defineProperty(exports, "red", { enumerable: true, get: function () { return kleur_1.red; } });
Object.defineProperty(exports, "blue", { enumerable: true, get: function () { return kleur_1.blue; } });
Object.defineProperty(exports, "green", { enumerable: true, get: function () { return kleur_1.green; } });
Object.defineProperty(exports, "italic", { enumerable: true, get: function () { return kleur_1.italic; } });
Object.defineProperty(exports, "yellow", { enumerable: true, get: function () { return kleur_1.yellow; } });
Object.defineProperty(exports, "bold", { enumerable: true, get: function () { return kleur_1.bold; } });
Object.defineProperty(exports, "dim", { enumerable: true, get: function () { return kleur_1.dim; } });
Object.defineProperty(exports, "magenta", { enumerable: true, get: function () { return kleur_1.magenta; } });
Object.defineProperty(exports, "black", { enumerable: true, get: function () { return kleur_1.black; } });
Object.defineProperty(exports, "white", { enumerable: true, get: function () { return kleur_1.white; } });
Object.defineProperty(exports, "gray", { enumerable: true, get: function () { return kleur_1.gray; } });
const color = (str, color) => {
    const c = (0, color_1.toRGB)(color);
    return `\x1B[38;2;${c.r};${c.g};${c.b}m${str}\x1B[39m`;
};
exports.color = color;
