// @sil/sass/dist/convert.test.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var convert_1 = require("./convert");
var sassValues = [
    { input: "rgba()", output: '"rgba()"' },
    { input: "rgba(0,0,0,0)", output: "rgba(0,0,0,0)" },
    { input: "#ffffff", output: "#ffffff" },
    { input: "true", output: "true" },
    { input: true, output: "true" },
    { input: "false", output: "false" },
    { input: false, output: "false" },
    { input: "ugh", output: '"ugh"' },
];
describe("SassValue", function () {
    Object.keys(sassValues).forEach(function (_v, key) {
        it("should convert to a Sass value", function () {
            expect((0, convert_1.toSassValue)(sassValues[key].input)).toBe(sassValues[key].output);
        });
    });
});
describe("SassObject", function () {
    it("Should convert to a Sass Object", function () {
        var input = {
            SomethingGoingOn: true,
            myValue: [1, 3, 5, "string"],
            left: "left",
            right: "center",
            position: "absolute",
            content: "something"
        };
        var output = "\"SomethingGoingOn\": true,\n\"myValue\": (1, 3, 5, string),\n\"left\": left,\n\"right\": center,\n\"position\": absolute,\n\"content\": \"something\"";
        expect((0, convert_1.toSassObject)(input)).toBe(output);
    });
});
describe("SassVariables", function () {
    it("Should convert to a Sass Variables", function () {
        var input = {
            SomethingGoingOn: true,
            myValue: [1, 3, 5, "string"],
            left: "left",
            right: "center",
            position: "absolute",
            content: "something"
        };
        var output = "$SomethingGoingOn: true;\n$myValue: (1, 3, 5, string);\n$left: left;\n$right: center;\n$position: absolute;\n$content: \"something\";";
        expect((0, convert_1.toSassVariables)(input)).toBe(output);
    });
});
